src/WindowsStateTriggers/AdaptiveTrigger.cs

```diff
@@ -0,0 +1,257 @@
+  /// The built-in AdaptiveTrigger apparently does some magic to determine it's range based upon other AdaptiveTrigger value in the same VisualState group
+  /// I could not repro that magic, so this trigger uses both max and min values
+  /// </summary>
+  public class AdaptiveTrigger : StateTriggerBase, ITriggerValue
+  {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AdaptiveTrigger"/> class.
+    /// </summary>
+    public AdaptiveTrigger()
+    {
+      var window = CoreApplication.GetCurrentView()?.CoreWindow;
+      if (window != null)
+      {
+        var weakEvent = new WeakEventListener<AdaptiveTrigger, CoreWindow, WindowSizeChangedEventArgs>(this)
+        {
+          OnEventAction = (instance, s, e) => OnCoreWindowOnSizeChanged(s, e),
+          OnDetachAction = (instance, weakEventListener) => window.SizeChanged -= weakEventListener.OnEvent
+        };
+        window.SizeChanged += weakEvent.OnEvent;
+      }
+    }
+
+    private void OnCoreWindowOnSizeChanged(CoreWindow sender, WindowSizeChangedEventArgs args)
+    {
+      OnCoreWindowOnSizeChanged(args.Size);
+    }
+
+    private void OnCoreWindowOnSizeChanged(Size size)
+    {
+      IsActive = size.Height >= MinWindowHeight && size.Width >= MinWindowWidth &&
+                 size.Height < MaxWindowHeight && size.Width < MaxWindowWidth &&
+                 MinWindowHeight < MaxWindowHeight && MinWindowWidth < MaxWindowWidth;
+    }
+
+    private void OnWindowHeightPropertyChanged()
+    {
+      var window = CoreApplication.GetCurrentView()?.CoreWindow;
+      if (window != null)
+      {
+        IsActive = window.Bounds.Height >= MinWindowHeight &&
+                   window.Bounds.Height < MaxWindowHeight &&
+                   MinWindowHeight < MaxWindowHeight;
+      }
+    }
+
+    private void OnWindowWidthPropertyChanged()
+    {
+      var window = CoreApplication.GetCurrentView()?.CoreWindow;
+      if (window != null)
+      {
+        IsActive = window.Bounds.Width >= MinWindowWidth &&
+                   window.Bounds.Width < MaxWindowWidth &&
+                   MinWindowWidth < MaxWindowWidth;
+      }
+    }
+
+    private bool _isActive;
+
+    /// <summary>
+    /// Gets a value indicating whether this trigger is active.
+    /// </summary>
+    /// <value><c>true</c> if this trigger is active; otherwise, <c>false</c>.</value>
+    public bool IsActive
+    {
+      get
+      {
+        return _isActive;
+      }
+      private set
+      {
+        if (_isActive != value)
+        {
+          _isActive = value;
+          base.SetActive(value);
+          IsActiveChanged?.Invoke(this, EventArgs.Empty);
+        }
+      }
+    }
+
+    /// <summary>
+    /// Occurs when the <see cref="IsActive" /> property has changed.
+    /// </summary>
+    public event EventHandler IsActiveChanged;
+
+
+    /// <summary>
+    /// Max size for a screen
+    /// </summary>
+    public static readonly double MaxSize = int.MaxValue;
```

Comment:
Shouldn't be public. Also I don't see why you need this - int.MaxValue is a constant that can be used already (and also no UI element supports this size - I believe the max size is actually 32762 last I checked)